haroonsheikh/HAROON_SHEIKH.md

```diff
@@ -0,0 +1,47 @@
+
+## Knowing the technology stack
+
+One area that has helped in getting most amount of test coverage is an understanding of the technology stack without knowing anything about coding.  I have This enables targeted testing of not only front-end components but also backend technologies including databases.
+
+- It really aids in understanding the inputs and outputs of the system and can easily see where in the automation pyramid a particular test needs to reside.  I.e., Do I really need a test at UI (User Interface) layer when it can easily be covered by a more stable integration/unit test.
+- It's important to understand how we can monitor the application behaviour by looking at the logs when an unexpected behaviour occurs at each layer of the stack and helps in tracing the flow or sequence through a system.
+- Just knowing the technology can help with identifying potentially interesting variables to explore with testing.  For example, the version of a particular technology used can help with picking on any present security vulnerabilities.  If you're looking to automate testing, then this can also help in deciding what tools and languages to choose for the tests. E.g., If the front-end application is written is `react-js` then writing the integration tests in `testcafe` maybe the preferred to stay in sync with the programming language and shareability of the selectors, etc.
+
+## Knowing the dependencies
+
+Another area of understanding that's helped me shape better tests and healthy conversation with other engineers is the knowledge of dependencies that a system may have on another system.  It could be dependencies within code, between two components or within different requirements.  One way to grasp the connections is by drawing out diagrams of different components of the system, and the communications between them, including the user journeys/interactions between them.  These technical/architecture diagrams can be used for identifying risks, highlight any potential issues earlier in the process, come up with tests that need carrying out, and start thinking about what tools might aid in testing.
+
+When I'm not sure about how a particular feature hangs together technically, then I just ask the developer next to me.
+
+## Pairing with developers
+
+One of most crucial practices of being a technical tester is the ability to pair with the developers.  This does not mean the ability to code but the ability to **read** code.  I like to pair with developers so that we're able to discuss the changes early and have a continuous 3-amigo style conversation throughout to build a more maintainable and testable solution.  I'm able to share my testing perspective to help developers think about those unhappy paths when writing unit tests.  This also encourages discussions around what tests should be covered at which layer of the automation pyramid without repeating ourselves.
```

Comment:
```suggestion
One of crucial practices of being a technical tester is the ability to pair with the developers.  This does not mean the ability to code but the ability to **read** code.  I like to pair with developers so that we're able to discuss the changes early and have a continuous 3-amigo style conversation throughout to build a more maintainable and testable solution.  I'm able to share my testing perspective to help developers think about those unhappy paths when writing unit tests.  This also encourages discussions around what tests should be covered at which layer of the automation pyramid without repeating ourselves.
```